src/function/scalar/nested/map/cardinality.cpp

```diff
@@ -8,22 +8,27 @@ namespace duckdb {
 
 static void CardinalityFunction(DataChunk &args, ExpressionState &state, Vector &result) {
 	auto &map = args.data[0];
-	result.SetVectorType(VectorType::CONSTANT_VECTOR);
+	result.SetVectorType(VectorType::FLAT_VECTOR);
 	auto result_data = FlatVector::GetData<uint64_t>(result);
 	if (!StructVector::HasEntries(map)) {
 		result_data[0] = 0;
 		return;
 	}
 	auto &children = StructVector::GetEntries(map);
-	idx_t map_size = ListVector::GetListSize(*children[0].second);
-	result_data[0] = map_size;
+	for (idx_t row = 0; row < args.size(); row++) {
```

Comment:
This needs to be orrified or flattened, or needs special handling depending on the vector type. As it is this loop will break if you have a constant vector with a size > 1, which can be generated by e.g. the following query:

```sql
select cardinality(m) from (select MAP(list_value(1), list_value(2)) from range(64) tbl(i)) tbl(m);
```